sql: panic on `SELECT MAX(true::bool)`

Found with RSG: https://teamcity.cockroachdb.com/viewLog.html?buildId=245247&buildTypeId=Cockroach_Nightlies_RandomSyntaxTests

```
[TestRandomSyntaxFunctions] [Test Output]
panic: interface conversion: interface is nil, not parser.TypedExpr [recovered]
	panic: SELECT MAX(true::bool): interface conversion: interface is nil, not parser.TypedExpr [recovered]
	panic: SELECT MAX(true::bool): interface conversion: interface is nil, not parser.TypedExpr [recovered]
	panic: SELECT MAX(true::bool): interface conversion: interface is nil, not parser.TypedExpr

goroutine 5678959 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc4311a0000, 0x7f63e27d8448, 0xc440d96060)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:200 +0xb1
panic(0x18872e0, 0xc46c2d2290)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve.func1(0xc44ee60c00, 0x7f63e27d8448, 0xc4262e8c30)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:353 +0x74
panic(0x18872e0, 0xc46c2d2290)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements.func1(0xc42043d0a7, 0x16)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:499 +0x131
panic(0x18bbc00, 0xc46d810b40)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/sql/parser.(*ComparisonExpr).memoizeFn(0xc457af2700)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/expr.go:396 +0xb2
github.com/cockroachdb/cockroach/pkg/sql/parser.NewTypedComparisonExpr(0x15, 0x0, 0x0, 0x26bd0e0, 0x2b2d220, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/expr.go:390 +0x106
github.com/cockroachdb/cockroach/pkg/sql.(*groupNode).addIsNotNullFilter(0xc466c72b40, 0xc4203a4000, 0xc4395bd9e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/group.go:415 +0xfd
github.com/cockroachdb/cockroach/pkg/sql.(*planner).SelectClause(0xc43094d980, 0x7f63e27d8448, 0xc4595903c0, 0xc457af2680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:294 +0x47f
github.com/cockroachdb/cockroach/pkg/sql.(*planner).Select(0xc43094d980, 0x7f63e27d8448, 0xc4595903c0, 0xc459590390, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:196 +0x50f
github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan(0xc43094d980, 0x7f63e27d8448, 0xc4595903c0, 0x26b7c20, 0xc459590390, 0x0, 0x0, 0x0, 0x6c1473, 0x59116144, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:340 +0x28db
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan(0xc43094d980, 0x7f63e27d8448, 0xc4595903c0, 0x26b7c20, 0xc459590390, 0x59116144, 0xebf25b0, 0xc42104c228, 0xc42104c2a8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:214 +0x8f
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420236fc0, 0x26b7c20, 0xc459590390, 0xc43094d980, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1554 +0x118
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420236fc0, 0xc45b524d80, 0x26b7c20, 0xc459590390, 0x0, 0x101, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1314 +0x463
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtsInCurrentTxn(0xc420236fc0, 0xc45b524d80, 0xc46c2d2150, 0x1, 0x1, 0x0, 0x101, 0x0, 0x0, 0xc42104cbf8, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:955 +0x818
github.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420236fc0, 0xc45b524d80, 0xc46c2d2150, 0x1, 0x1, 0x0, 0x1, 0xc46c2d2168, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:863 +0xd1
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest.func1(0x7f63e27d8448, 0xc4595903c0, 0xc420524c60, 0xc46c2d2168, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:695 +0x1df
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc420524c60, 0x7f63e27d8448, 0xc4595903c0, 0x10101, 0xc465acc070, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:640 +0xb6
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420236fc0, 0xc45b524d80, 0xc42043d0a7, 0x16, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc42043d000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:717 +0x66d
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements(0xc420236fc0, 0xc45b524d80, 0xc42043d0a7, 0x16, 0x0, 0x0, 0x0, 0x0, 0xc42043d000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:505 +0x11b
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).executeStatements(0xc44ee60c00, 0xc42043d0a7, 0x16, 0x0, 0x0, 0x0, 0x0, 0xa79601, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:836 +0x94
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).handleSimpleQuery(0xc44ee60c00, 0xc44ee60c28, 0xc46f48f020, 0x51)
```

--- a/pkg/sql/group.go
+++ b/pkg/sql/group.go
@@ -474,7 +474,7 @@ func (n *groupNode) addIsNotNullFilter(where *filterNode, render *renderNode) {
 		where.ivarHelper.Rebind(
 			render.render[n.desiredOrdering[0].ColIdx],
 			false, // alsoReset
-			false, // normalizeToNonNil
+			true,  // normalizeToNonNil
 		),
 		parser.DNull,
 	)

--- a/pkg/sql/testdata/logic_test/aggregate
+++ b/pkg/sql/testdata/logic_test/aggregate
@@ -1263,3 +1263,9 @@ SELECT TO_HEX(XOR_AGG(a)), b, XOR_AGG(c) FROM xor_bytes GROUP BY b ORDER BY b
 
 statement error arguments to xor must all be the same length
 SELECT XOR_AGG(i) FROM (VALUES (b'\x01'), (b'\x01\x01')) AS a(i)
+
+query BB
+SELECT MAX(true), MIN(true)
+----
+true
+true

